manifests/stable/vertical-pod-autoscaler.clusterserviceversion.yaml

```diff
@@ -547,8 +547,6 @@ spec:
                   runAsNonRoot: true
                   seccompProfile:
                     type: RuntimeDefault
-              nodeSelector:
-                node-role.kubernetes.io/master: ""
```

Comment:
If we're taking this out of the CSV...in a case where we're not overriding, does the nodeSelector get defaulted to "master" somewhere else in the code/somehow ?

(Like, I assume we want to preserve existing "masters only" behavior for folks who aren't overriding, or do we not?)

Reply:
This is a change in behavior for this feature. It allows the VPA operator to be scheduled anywhere on all clusters. The tolerations are retained so that it can, but is not required to, run on masters. Unfortunately, I'm not aware of a way that we can allow the cluster admin to control where it lands by way of OLM.

Reply:
Perhaps we should use a node affinity with preferredDuringSchedulingIgnoredDuringExecution so that in clusters where it can run on a master, it will. And we could have a high-weight rule which looks for a vpa-specific node label so that they can label the node where they want the VPA to land, if they really care to control that.

Reply:
Ok cool, thanks. I think that's alright, it's just the operator, if it dies the operand stays alive.

Reply:
It looks like they have a way to select the operator node via the subscription's spec.config.nodeSelector field. I'll investigate further.

Reply:
John and I talked it over. We're going to revert the above change, and rely upon an OLM user's ability to override the nodeselector via the subscription config, as explained here:
https://github.com/operator-framework/operator-lifecycle-manager/blob/master/doc/design/subscription-config.md#nodeselector
That means a cluster admin on a hypershift cluster will need to take the extra step of specifying where the VPA operator should land. We will document this extra step in the OpenShift docs